libp2p/src/main/kotlin/io/libp2p/pubsub/gossip/GossipRouter.kt

```diff
@@ -459,6 +475,8 @@ open class GossipRouter(
                 .whenTrue { notifyIWantTimeout(key.first, key.second) }
         }
 
+        peerIDontWant.clear()
```

Comment:
We may clear remote `IDONTWANT` which is still in effect here. We probably need some more sophisticated pruning mechanism here